app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsBloomFilterFactory.kt

```diff
@@ -16,43 +16,39 @@
 
 package com.duckduckgo.app.httpsupgrade.api
 
+import android.support.annotation.WorkerThread
 import com.duckduckgo.app.global.store.BinaryDataStore
 import com.duckduckgo.app.httpsupgrade.BloomFilter
 import com.duckduckgo.app.httpsupgrade.db.HttpsBloomFilterSpecDao
 import com.duckduckgo.app.httpsupgrade.model.HttpsBloomFilterSpec.Companion.HTTPS_BINARY_FILE
-import io.reactivex.Observable
-import io.reactivex.schedulers.Schedulers
 import timber.log.Timber
+import java.util.*
 import javax.inject.Inject
 
 interface HttpsBloomFilterFactory {
     fun create(): BloomFilter?
 }
 
-class OptionalWrapper<out T>(val value: T?)
 
 class HttpsBloomFilterFactoryImpl @Inject constructor(private val dao: HttpsBloomFilterSpecDao, private val binaryDataStore: BinaryDataStore) :
     HttpsBloomFilterFactory {
 
+    @WorkerThread
     override fun create(): BloomFilter? {
 
-        val specificationWrapper = Observable.just(dao)
-            .observeOn(Schedulers.io())
-            .map {
-                OptionalWrapper(dao.get())
-            }
-            .blockingSingle()
-
-        val specification = specificationWrapper.value
+        val specification = dao.get()
         val dataPath = binaryDataStore.dataFilePath(HTTPS_BINARY_FILE)
 
         if (dataPath == null || specification == null) {
             Timber.d("Https update data not found")
             return null
         }
 
+        val initialTimestamp = Date().time
```

Comment:
Given this isn't performance-critical code path, this probably doesn't matter but....

creating a new `Date` object here is more overhead than grabbing the `System.currentTimeMillis()` directly

in fact, the constructor of `Date` does just that anyway

    public Date() {
        this(System.currentTimeMillis());
    }

so we'd be better replacing `Date().time` with `System.currentTimeMillis()` I think.